### PR TITLE
refactor(ui): move to /ui folder, switch to vite for build

### PR DESCRIPTION
**Pull request by davinov**
Part of #912 was to move everything UI-related into its own folder, so the root stays cleans.
I took this opportunity to also switch the build system to vite (it became necessary because typescript needed to be updated too, and the rollup plugins were not following).

Sorry for this faaaaaaat PR, tat contained a lot of sed/replacements in order to clean all the build errors.

TODO:
- finish cleaning error of types build
- ~~repair storybook~~
- ~~switch tests to vitest~~
- deduplicate playground code
- ~~adapt github actions and sonar~~

### CODE REVIEW

**Comment by davinov** [comment]
OK, I think it's good enough for now. I prefer merging this ASAP to unblock the ability to do releases.
Need to keep in mind that:
- A lot of typechecking errors are still there, there were not previously visible because the CI doesn't check for them. the migration to vue 2.7 also improved the typechecking.
- A few tests were skipped (popover, filter editor). Their components works OK, I just hadn't had the time to figure their problem (which doesn't seem trivial).
- Only some stories were migrated, there is still many to do!

File: .gitignore [hunk 1]
```diff
@@ -1,25 +1,8 @@
-node_modules
-/dist
-.storybook/dist
-playground/*.map
-
+# Editors
 .idea
```
Comment (jGundermann) [inline .gitignore:2]: This should maybe stay here

```suggestion
.idea
.vscode
```

**Review by jGundermann — COMMENTED** [review]
Thx for the great PR! Huge work congrats. 
Got some type error during build. and found a minor change to make.

File: ui/src/components/stepforms/widgets/Autocomplete.vue [hunk 2]
```diff
@@ -60,7 +60,7 @@ import VariableInput from './VariableInput.vue';
     FAIcon,
   },
 })
-export default class AutocompleteWidget extends Mixins(FormWidget) {
+export default class AutocompleteWidget extends FormWidget {
```
Comment (davinov) [inline ui/src/components/stepforms/widgets/Autocomplete.vue:63]: Ah oui t'as carrément enlever l'usage de Mixins! Bonne idée ^^

**Review by bloodstorms — APPROVED** [review]
Tout OK pour moi, testé en local super fast 🚀, tout fonctionne comme avant, pas d'erreur en console 👌 
Let's meeeeeeerge

**Pull request merged**